plugins/trivy/cmd/trivy/main.go

```diff
@@ -37,37 +37,40 @@ type newestVersions struct {
  */
 func main() {
 	setEnv()
-	lastReport := image.GetLastReport()
+	lastReport, err := image.GetLastReport()
+	if err != nil {
+		logrus.Fatal(err)
+	}
 	ctx := context.Background()
 	images, err := image.GetImages(ctx)
 	if err != nil {
-		panic(err)
+		logrus.Fatal(err)
 	}
 	logrus.Infof("Listing images from cluster:")
 	for _, i := range images {
 		logrus.Infof("%v - %v", i.ID, i.Name)
 	}
 	imagesToScan := getUnscannedImagesToScan(images, lastReport.Images)
-	imagesToScan = getImagesToRescan(images, lastReport, imagesToScan)
+	imagesToScan = getImagesToRescan(images, *lastReport, imagesToScan)
 	logrus.Infof("Listing images to be scanned:")
 	for _, i := range imagesToScan {
 		logrus.Infof("%v - %v", i.ID, i.Name)
 	}
-	clusterImagesToKeep := getClusterImagesToKeep(images, lastReport, imagesToScan)
+	clusterImagesToKeep := getClusterImagesToKeep(images, *lastReport, imagesToScan)
 	allReports := image.ScanImages(imagesToScan, maxConcurrentScans, extraFlags, false)
 	recommendationsToScan := getNewestVersionsToScan(ctx, allReports, imagesToScan)
 	recommendationReport := image.ScanImages(recommendationsToScan, maxConcurrentScans, extraFlags, true)
-	recommendationsToKeep := getRecommendationImagesToKeep(images, lastReport, recommendationsToScan)
+	recommendationsToKeep := getRecommendationImagesToKeep(images, *lastReport, recommendationsToScan)
 	lastReport.Images = append(clusterImagesToKeep, recommendationsToKeep...)
 	aggregated := append(allReports, recommendationReport...)
-	finalReport := image.Minimize(aggregated, lastReport)
-	data, err := json.Marshal(finalReport)
+	minimizedReport := image.Minimize(aggregated, *lastReport)
+	data, err := json.Marshal(minimizedReport)
 	if err != nil {
-		panic(err)
+		logrus.Fatalf("could not marshal report: %v", err)
```

Comment:
neat. did not know about this.

Reply:
yep! It is basically `log error + os.exit(1)`